hbase-server/src/main/java/org/apache/hadoop/hbase/namespace/NamespaceStateManager.java

```diff
@@ -186,6 +197,9 @@ void deleteNamespace(String namespace) {
   }
 
   private void addTable(TableName tableName, int regionCount) throws IOException {
+    if (tableName.isSystemTable()) {
```

Comment:
If these add/remove private methods are only called by the others, we may not need to add the check here since we already guard the callers. But I didn't trace all the calls and also the redundancy is not a huge issue. Just calling it out.